src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -1268,6 +1268,70 @@ public Void invoke(Repository repo, VirtualChannel channel) throws IOException,
             }
         });
     }
+    
+    /**
+     * Verifies that if project specifies LocalBranch with value of "**" 
+     * that the checkout to a local branch using remote branch name sans 'origin'.
+     * This feature is necessary to support Maven release builds that push updated
+     * pom.xml to remote branch as 
+     * <br/>
+     * <pre>
+     * git push origin localbranch:localbranch
+     * </pre>
+     * @throws Exception
+     */
+    public void testCheckoutToDefaultLocalBranch_StarStar() throws Exception {
+       FreeStyleProject project = setupSimpleProject("master");
+
+       final String commitFile1 = "commitFile1";
+       commit(commitFile1, johnDoe, "Commit number 1");
+       GitSCM git = (GitSCM)project.getScm();
+       git.getExtensions().add(new LocalBranch("**"));
+       FreeStyleBuild build1 = build(project, Result.SUCCESS, commitFile1);
+
+       assertEquals("GIT_BRANCH", "origin/master", getEnvVars(project).get(GitSCM.GIT_BRANCH));
+       assertEquals("GIT_LOCAL_BRANCH", "master", getEnvVars(project).get(GitSCM.GIT_LOCAL_BRANCH));
+    }
+
+    /**
+     * Verifies that if project specifies LocalBranch with null value (empty string) 
+     * that the checkout to a local branch using remote branch name sans 'origin'.
+     * This feature is necessary to support Maven release builds that push updated
+     * pom.xml to remote branch as 
+     * <br/>
+     * <pre>
+     * git push origin localbranch:localbranch
+     * </pre>
+     * @throws Exception
+     */
+    public void testCheckoutToDefaultLocalBranch_NULL() throws Exception {
+       FreeStyleProject project = setupSimpleProject("master");
+
+       final String commitFile1 = "commitFile1";
+       commit(commitFile1, johnDoe, "Commit number 1");
+       GitSCM git = (GitSCM)project.getScm();
+       git.getExtensions().add(new LocalBranch(""));
+       FreeStyleBuild build1 = build(project, Result.SUCCESS, commitFile1);
+
+       assertEquals("GIT_BRANCH", "origin/master", getEnvVars(project).get(GitSCM.GIT_BRANCH));
+       assertEquals("GIT_LOCAL_BRANCH", "master", getEnvVars(project).get(GitSCM.GIT_LOCAL_BRANCH));
+    }
+
+    /**
+     * Verifies that GIT_LOCAL_BRANCH is not set if LocalBranch extension
+     * is not configured.
+     * @throws Exception
+     */
+    public void testCheckoutSansLocalBranchExtension() throws Exception {
+       FreeStyleProject project = setupSimpleProject("master");
+
+       final String commitFile1 = "commitFile1";
+       commit(commitFile1, johnDoe, "Commit number 1");
+       FreeStyleBuild build1 = build(project, Result.SUCCESS, commitFile1);
+
+       assertEquals("GIT_BRANCH", "origin/master", getEnvVars(project).get(GitSCM.GIT_BRANCH));
+       assertEquals("GIT_LOCAL_BRANCH", null, getEnvVars(project).get(GitSCM.GIT_LOCAL_BRANCH));
+    }
 
```

Comment:
The above testCheckoutSansLocalBranchExtension fails when build is run in Jenkins AND the the LocalBranch extension is configured for the job.  I suspect the test case is seeing the LocalBranch for the job that is running the test instead of the test project itself.  Is this a defect in the test infrastructure allowing the Jenkins job configuration to bleed into the test case, or is my test  case incorrect?  Is there something I need to include to completely isolate the test case from the build environment?

Following log message from my local jenkins build.
testCheckoutSansLocalBranchExtension(hudson.plugins.git.GitSCMTest)  Time elapsed: 1.316 sec  <<< FAILURE!
junit.framework.ComparisonFailure: GIT_LOCAL_BRANCH expected:<null> but was:<master>
	at hudson.plugins.git.GitSCMTest.testCheckoutSansLocalBranchExtension(GitSCMTest.java:1333)